File: atcoder/abc/python/abc215/abc215_D.py

import sys
infile = sys.stdin.buffer

def gs()  : return infile.readline().rstrip()
def gi()  : return int(gs())
def gf()  : return float(gs())
def gss() : return gs().split()
def gis() : return [int(x) for x in gss()]
def gfs() : return [float(x) for x in gss()]

def fsieve(n) :
    res = [-1] * (n+1)
    for k in range(2,n+1,2) : res[k] = 2
    for i in range(3,n+1,2) :
        if res[i] > 0 : continue
        res[i] = i
        for k in range(i*i,n+1,2*i) : 
            if res[k] == -1 : res[k] = i
    return res

def main(infn="") :
    global infile
    infile = open(infn,"r") if infn else open(sys.argv[1],"r") if len(sys.argv) > 1 else sys.stdin
    N,M = gis()
    A = gis()
    factors = fsieve(100_001)
    pfactors = [False] * (100_001)
    for a in A :
        aa = a
        while aa > 1 :
            p = factors[aa]
            pfactors[p] = True
            while aa % p == 0 : aa //= p
    sb = [False] * (100_001); ans = [1]; sb[1] = True
    for i in range(2,M+1) :
        p = factors[i]
        if not pfactors[p] and sb[i//p] : ans.append(i); sb[i] = True
    ansstr = "\n".join([str(x) for x in ans])
    sys.stdout.write(str(len(ans))+'\n')
    sys.stdout.write(str(ansstr)+'\n')

if __name__ == '__main__' :
    main()
    sys.stdout.flush()

File: abc/go/abc215/abc215_D/abc215_D.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)
var wrtr = bufio.NewWriterSize(os.Stdout, 10_000_000)
var rdr = bufio.NewScanner(os.Stdin)
func gs() string  { rdr.Scan(); return rdr.Text() }
func gi() int     { i,e := strconv.Atoi(gs()); if e != nil {panic(e)}; return i }
func gi2() (int,int) { return gi(),gi() }
func gis(n int) []int  { res := make([]int,n); for i:=0;i<n;i++ { res[i] = gi() }; return res }
func iai(m int,v int) []int { a := make([]int,m); for i:=0;i<m;i++ { a[i] = v }; return a }

func fsieve(n int) []int {
	s := iai(n+1,-1)
	s[0] = 0; s[1] = 1; for i:=2;i<=n;i+=2 { s[i] = 2 }
	for i:=3;i<=n;i+=2 { 
		if s[i] != -1 { continue }
		s[i] = i
		for k:=i*i;k<=n;k+=2*i { if s[k] == -1 { s[k] = i } }
	}
	return s
}

func main() {
	//f1, _ := os.Create("cpu.prof"); pprof.StartCPUProfile(f1); defer pprof.StopCPUProfile()
	defer wrtr.Flush()
	infn := ""; if infn == "" && len(os.Args) > 1 {	infn = os.Args[1] }
	if infn != "" {	f, e := os.Open(infn); if e != nil { panic(e) }; rdr = bufio.NewScanner(f) }
	rdr.Split(bufio.ScanWords); rdr.Buffer(make([]byte,1024),1_000_000_000)
	// PROGRAM STARTS HERE
	N,M := gi2(); A := gis(N)
	f := fsieve(100_000)
	pfactors := make([]bool,100_001)
	for _,a := range A {
		aa := a
		for aa != 1 {
			p := f[aa];
			pfactors[p] = true
			for aa % p == 0 { aa /= p } 
		}
	}
	sb := make([]bool,100_001); ans := []int{1}; sb[1] = true
	for i:=2;i<=M;i++ { 
		p := f[i]
		if !pfactors[p] && sb[i/p] { ans = append(ans,i); sb[i] = true }
	}
	fmt.Fprintln(wrtr,len(ans))
	for _,a := range ans { fmt.Fprintln(wrtr,a) }
}
